core/types/block.go

```diff
@@ -104,6 +105,9 @@ type Header struct {
 
 	//Hacky: used internally to mark the header as requested by the downloader at the deliver queue
 	Requested bool `json:"-" rlp:"-"`
+	//Hacky: used internally store the signature in memory.
+	BlockSignature []byte `json:"-" rlp:"-"`
+	EuclidHandled  bool   `json:"-" rlp:"-"`
```

Comment:
Can't we infer if it was handled by just checking if `BlockSignature == nil`?

Reply:
Deprecated. `BlockSignature == nil ` is the default value, whereas `EuclidHandled = true` is not.